src/modules/admin/admin.service.ts

```diff
@@ -0,0 +1,16 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/common/prisma/prisma.service';
+import { CreateRoleDto } from './dto/create-role.dto';
+
+@Injectable()
+export class AdminService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  async createRole({ name }: CreateRoleDto) {
+    const role = await this.prismaService.role.findUnique({ where: { name } });
+    if (role) {
+      throw new BadRequestException(`Role: ${role.name} already exists`);
+    }
+    return this.prismaService.role.create({ data: { name } });
```

Comment:
Is there a point to return something in this function?

Reply:
idk, will be changed later, did used this module.

Reply:
It depends, nut usually create endpoint does not return anything